pypsa/components.py

```diff
@@ -862,14 +911,18 @@ def copy(self, with_time=True, snapshots=None, ignore_standard_types=False):
         if with_time:
             if snapshots is None:
                 snapshots = self.snapshots
+            if investment_periods is None:
+                investment_periods = self.investment_period_weightings.index
```

Comment:
also here, not using `investment_period_weightings.index` but `investment_periods`

Reply:
```suggestion
                investment_periods = self.investment_periods
```